source/utilities.cc

```diff
@@ -221,15 +226,15 @@ namespace aspect
                                  + "The current structure vector has " + std::to_string(n_values_per_key->size()) + " entries, but there are "
                                  + std::to_string(n_fields) + " field names." ));
 
-        for (const auto &entry: parsed_map)
+        for (const std::pair<std::string, double> &key_and_value: parsed_map)
           {
-            const auto field_name = std::find(field_names.begin(),field_names.end(),entry.first);
+            const auto field_name = std::find(field_names.begin(),field_names.end(),key_and_value.first);
```

Comment:
Just for consistency, as you changed it in all the other places too: Maybe change this one too: 
`auto` --> `const std::string`

Reply:
unfortunately this is not a `std::string`, it is a `std::vector<std::string>::iterator`. Still want me to change it?

Reply:
Hm. I am not sure. 
But maybe just the fact that I didn't know the return type of the functions means changing it would make the code clearer?